gpii/node_modules/flowManager/test/SolutionsRegistryDataSourceTests.js

```diff
@@ -0,0 +1,320 @@
+
+// ======== Testing the solutions registry data source used by the LFM ========
+
+// Test component for the LFM solutions registry data source mocking a load
+// from the source code repository.
+fluid.defaults("gpii.tests.solutionsRegistry.dataSource.local.mockLoadFromRepository", {
+    gradeNames: ["gpii.flowManager.solutionsRegistry.dataSource.local"],
+    components: {
+        revisionRequester: {
+            type: "gpii.flowmanager.revisionRequester",
+            options: {
+                cloudURL: "http://gpii.net"
+            }
+        }
+    },
+    listeners: {
+        "loadSolutions.loadFromLocalDisk": {
+            listener: "{that}.loadFromLocalDisk",
+            priority: "first"
+        },
+        "loadSolutions.getRevision": {
+            listener: "{revisionRequester}.getRevision",
+            priority: "after:loadFromLocalDisk"
+        },
+        "loadSolutions.loadFromRepository": {
+            listener: "gpii.tests.solutionsRegistry.dataSource.local.mockRepoLoad",
+            args: [
+                "{that}",
+                gpii.tests.solutionsRegistry.platformId,
+                gpii.tests.solutionsRegistry.repositoryDarwinSolutions
+            ],
+            priority: "after:getRevision"
+        },
+        "loadSolutions.solutionsLoaded": {
+            listener: "{that}.events.solutionsRegistryReady.fire",
+            priority: "after:loadFromRepository"
+        }
+    }
+});
+
+gpii.tests.solutionsRegistry.dataSource.local.mockRepoLoad = function (that, platformId, solutions) {
+    var promise = fluid.promise();
+    var solutionsRegistry = {};
+    solutionsRegistry[platformId] = solutions;
+    that.repositorySolutionsRegistry = solutionsRegistry;
+    promise.resolve(that.repositorySolutionsRegistry);
+    return promise;
+};
+
+fluid.defaults("gpii.tests.solutionsRegistry.local.mockRepository.testCaseHolder", {
+    gradeNames: "fluid.test.testCaseHolder",
+    components: {
+        solutionsRegistryMockRepoLoad: {
+            type: "gpii.tests.solutionsRegistry.dataSource.local.mockLoadFromRepository",
+            options: {
+                path: gpii.tests.solutionsRegistry.path
+            }
+        }
+    },
+    modules: [{
+        name: "Solutions registry used by LFM - repository + file system based solutions",
+        expect: 2,
+        tests: [{
+            name: "Solutions registry LFM - solutions loaded from repository and looking for named registry",
+            sequence: [
+                { funcName: "fluid.log", args: ["Solutions Registry Mock Repo Load, getting named repository registry -- START"] },
+                {
+                    event: "{solutionsRegistryMockRepoLoad}.events.solutionsRegistryReady",
+                    listener: "jqUnit.assert",
+                    args: ["LFM solutionsRegistryMockRepoLoad ready"]
+                }, {
+                    funcName: "{solutionsRegistryMockRepoLoad}.get",
```

Comment:
Is the second `get()` call at line 187 needed? The result from this `get()` call can be verified for both: `darwin` content matches what's received from the repo and `win32` content matches what's in the local file system.

Reply:
Yes, it's needed.  The result of the `get()` at line 172 is just the `darwin` solutions.  Specifically, the promise returned by the `get()` resolves to an object containing all and only `darwin` solutions.  Nothing is returned about the `win32` content.  And, as you noted, this call to `get()` is to check that the `darwin` solutions are the ones retrieved from the source code repository, not the ones from the local file system.  I've modified the 
"check" function to make this more apparent.

The call to `get()` on line 187 is for checking that the `win32` solutions are returned, _and_ that they match the ones on the local file system.

That said, these tests are only somewhat meaningful.  They do check the logic of which solutions to return. But, by the time the `get()` requests are made, all of the solutions registries are in memory inside the SRDS.  Most test failures here point to prior failures of the solutions loading sequence.

Related:  Since the solutions are all in memory, why is this a GET request?  The result could be an immediate return of solutions rather than a fulfilled promise.  Is there a requirement that this `get()` be RESTful?  Perhaps @amb26 knows/remembers.